demosplan/DemosPlanCoreBundle/Command/Data/DeleteProcedureCommand.php

```diff
@@ -0,0 +1,660 @@
+    {
+        $this->deleteFromTableByIdentifierArray('annotated_statement_pdf_page', 'annotated_statement_pdf', $annotatedStatementPdfIds);
+    }
+
+    private function deleteAnnotatedStatementPdfs(): void
+    {
+        $this->deleteFromTableByIdentifierArray('annotated_statement_pdf', '_procedure', [$this->procedureId]);
+    }
+
+    private function deletePredefinedTextsCategoriesRelation(array $predefinedCategoryIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('predefined_texts_categories', '_ptc_id', $predefinedCategoryIds);
+    }
+
+    private function deletePredefinedTextsCategories(): void
+    {
+        $this->deleteFromTableByIdentifierArray('_predefined_texts_category', '_p_id', [$this->procedureId]);
+    }
+
+    private function deletePredefinedTexts(): void
+    {
+        $this->deleteFromTableByIdentifierArray('_predefined_texts', '_p_id', [$this->procedureId]);
+    }
+
+    private function deleteBoilerplateGroup(): void
+    {
+        $this->deleteFromTableByIdentifierArray('boilerplate_group', 'procedure_id', [$this->procedureId]);
+    }
+
+    private function deleteStatementImportEmailAttachments(array $importEmailIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('statement_import_email_attachments', 'statement_import_email_id', $importEmailIds);
+    }
+
+    private function deleteImportEmailOriginalStatements(array $importEmailIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('statement_import_email_original_statements', 'statement_import_email_id', $importEmailIds);
+    }
+
+    private function deleteStatementImportEmails(array $importEmailIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('statement_import_email', 'id', $importEmailIds);
+    }
+
+    private function deleteHashedQueries(): void
+    {
+        $this->deleteFromTableByIdentifierArray('hashed_query', 'procedure_id', [$this->procedureId]);
+    }
+
+    private function deleteTags(array $tagTopicIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('_tag', '_tt_id', $tagTopicIds);
+    }
+
+    private function deleteTagTopics(): void
+    {
+        $this->deleteFromTableByIdentifierArray('_tag_topic', '_p_id', [$this->procedureId]);
+    }
+
+    private function deleteProcedureNews(): void
+    {
+        $this->deleteFromTableByIdentifierArray('_news', '_p_id', [$this->procedureId]);
+    }
+
+    private function deleteStatementMeta(array $statementIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('_statement_meta', '_st_id', $statementIds);
+    }
+
+    private function deleteStatementAttachment(array $statementIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('statement_attachment', 'statement_id', $statementIds);
+    }
+
+    private function deleteFiles(array $fileIds): void
+    {
+        $this->deleteFromTableByIdentifierArray('_files', '_f_ident', $fileIds);
```

Comment:
Shouldn't these files also be removed physically?
```DemosPlanCoreBundle/Logic/FileService::deleteFile($hash)```

Reply:
Yes, I could do that. But if I remember correctly, we also have an orphaned files removal in our maintenance service/command/whatever. So I decided to leave the cleanup to that.

Reply:
Indeed and I would also go with the existing orphan removal in MaintenanceTask